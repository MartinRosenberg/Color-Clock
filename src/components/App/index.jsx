import moment from 'moment'
import React, { Component } from 'react'

const getNewState = () => {
  const time = moment()
  return {
    color: time.format('#HHmmss'),
    time: time.format('H:mm:ss'),
  }
}

export class App extends Component {
  state = getNewState()

  setClock = () => {
    this.setState(() => getNewState())
    document.body.setAttribute('style', `background: ${this.state.color}`)
  }

  componentDidMount () {
    this.setClock()
    this.tick = setInterval(this.setClock, 1000)
  }

  componentWillUnmount () {
    clearInterval(this.tick)
  }

  render () {
    return (
      <div style={{
        alignItems: 'center',
        color: '#eeeeee',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        height: '100vh',
        width: '100vw',
      }}>
        <h1>{this.state.time}</h1>
        <h2>{this.state.color}</h2>
      </div>
    )
  }
}
