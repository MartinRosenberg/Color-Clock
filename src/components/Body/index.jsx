import React from 'react'
import Helmet from 'react-helmet'

export const Body = ({ color }) => (
  <Helmet>
    <body style={`background: ${color}`}/>
  </Helmet>
)
