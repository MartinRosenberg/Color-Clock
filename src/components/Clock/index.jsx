import PropTypes from 'prop-types'
import React from 'react'

export const Clock = ({ color, time }) => (
  <div style={{
    alignItems: 'center',
    color: '#eeeeee',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100vh',
    width: '100vw',
  }}>
    <h1>{time}</h1>
    <h2>{color}</h2>
  </div>
)

Clock.propTypes = {
  time: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
}
