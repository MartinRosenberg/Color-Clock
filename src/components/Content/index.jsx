import PropTypes from 'prop-types'
import React from 'react'

import { style } from './constants'

export const Content = ({ children }) =>
  <div style={style}>
    {children}
  </div>

Content.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ])
}
