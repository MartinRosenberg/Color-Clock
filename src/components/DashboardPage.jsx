import React from 'react'

export const DashboardPage = () =>
  <div>
    <p>Dashboard page content</p>
  </div>
