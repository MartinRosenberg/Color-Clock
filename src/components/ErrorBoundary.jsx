import React from 'react'

export class ErrorBoundary extends React.Component {
  state = {
    error: undefined
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { error };
  }

  componentDidCatch(error, info) {
    // You can also log the error to an error reporting service
    console.log(error, info);
  }

  render() {
    if (this.state.error) {
      // You can render any custom fallback UI
      return <h4>Something went wrong.</h4>;
    }

    return this.props.children;
  }
}
