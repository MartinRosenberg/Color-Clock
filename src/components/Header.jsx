import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import { startLogOut } from '../actions/auth'

export const BaseHeader = ({ startLogOut }) =>
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Boilerplate React App</h1>
        </Link>
        <button className="button button--link" onClick={startLogOut}>
          Log Out
        </button>
      </div>
    </div>
  </header>

BaseHeader.propTypes = {
  startLogOut: PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  startLogOut: () => dispatch(startLogOut())
})

export const Header = connect(undefined, mapDispatchToProps)(BaseHeader)
