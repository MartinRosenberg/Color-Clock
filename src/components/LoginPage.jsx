import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'

import { startLogIn } from '../actions/auth'

export const BaseLoginPage = ({ startLogIn }) =>
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate React App</h1>
      <p>Tagline for app</p>
      <button className="button" onClick={startLogIn}>
        Log in with Google
      </button>
    </div>
  </div>

BaseLoginPage.propTypes = {
  startLogIn: PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  startLogIn: () => dispatch(startLogIn())
})

export const LoginPage = connect(undefined, mapDispatchToProps)(BaseLoginPage)
