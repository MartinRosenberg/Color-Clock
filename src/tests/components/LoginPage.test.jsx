import { shallow } from 'enzyme'
import React from 'react'

import { BaseLoginPage } from '../../components/LoginPage'

let startLogIn, wrapper

beforeEach(() => {
  startLogIn = jest.fn()
  wrapper = shallow(<BaseLoginPage startLogIn={startLogIn} />)
})

test('Should render LoginPage correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should call startLogIn on button click', () => {
  wrapper.find('button').prop('onClick')()
  expect(startLogIn).toHaveBeenCalled()
})
