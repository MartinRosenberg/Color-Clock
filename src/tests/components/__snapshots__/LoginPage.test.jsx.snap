// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate React App
    </h1>
    <p>
      Tagline for app
    </p>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Log in with Google
    </button>
  </div>
</div>
`;
